### PR TITLE
perf: avoid creating temporary docker image

### PR DESCRIPTION
**Pull request by paulpach**
when running the tests,  we create a brand new docker image with all the scripts inside.

This changes the approach to just mount the scripts as a volume.  In my self hosted runner this is significantly reducing the time it takes to test my app
Before:
Screen Shot 2022-03-30 at 1 43 02 PM
After:
Screen Shot 2022-03-30 at 1 43 47 PM

#### Changes

- ...

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

File: src/model/docker.ts [hunk 1]
```diff
@@ -72,11 +73,15 @@ const Docker = {
...
         --volume "${workspace}":"/github/workspace:z" \
+        --volume "${actionFolder}/steps":"/steps:z" \
+        --volume "${actionFolder}/entrypoint.sh":"/entrypoint.sh:z" \
         ${sshAgent ? `--volume ${sshAgent}:/ssh-agent` : ''} \
         ${sshAgent ? '--volume /home/runner/.ssh/known_hosts:/root/.ssh/known_hosts:ro' : ''} \
         ${useHostNetwork ? '--net=host' : ''} \
         ${githubToken ? '--env USE_EXIT_CODE=false' : '--env USE_EXIT_CODE=true'} \
-        ${image}`;
+        ${image} \
+        --
```
Comment (paulpach) [inline src/model/docker.ts:83]:
```suggestion
        -- \
```

File: src/model/docker.ts [hunk 2]
```diff
@@ -72,11 +73,15 @@ const Docker = {
...
+        --volume "${actionFolder}/steps":"/steps:z" \
+        --volume "${actionFolder}/entrypoint.sh":"/entrypoint.sh:z" \
         ${sshAgent ? `--volume ${sshAgent}:/ssh-agent` : ''} \
         ${sshAgent ? '--volume /home/runner/.ssh/known_hosts:/root/.ssh/known_hosts:ro' : ''} \
         ${useHostNetwork ? '--net=host' : ''} \
         ${githubToken ? '--env USE_EXIT_CODE=false' : '--env USE_EXIT_CODE=true'} \
-        ${image}`;
+        ${image} \
+        -- \
+        /usr/bin/bash /entrypoint.sh` ;
```
Comment (webbertakken) [inline src/model/docker.ts:84]: [](https://user-images.githubusercontent.com/20756439/160925541-594be8b0-b8f4-48ac-9903-855209de796b.png)I believe you don't need to pass the -- in this case, that's only necessary in k8s when passing something to the API and then passing an argument to the resulting resource.

In this case we're running a command in the docker image directly. So we can do the command part without the double hyphens

image

```suggestion
        ${image} \
        /usr/bin/bash /entrypoint.sh` ;
```

**Review by webbertakken — CHANGES_REQUESTED** [review]
Amazing performance fix! Thank you.

Looks mostly good!

**Review by webbertakken — APPROVED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -24,10 +24,10 @@ async function run() {
 
     try {
       // Build docker image
```
Comment (paulpach) [inline src/index.ts:26]:
```suggestion
```

File: src/index.ts [hunk 4]
```diff
@@ -23,11 +23,10 @@ async function run() {
     const runnerTempPath = process.env.RUNNER_TEMP;
 
     try {
-      // Build docker image
-      const actionImage = await Docker.build({ path: actionFolder, dockerfile, baseImage });
 
       // Run docker image
```
Comment (webbertakken) [inline src/index.ts:27]:
```suggestion
```

**Comment by webbertakken** [comment]
Thank you very much for this nice performance improvement!

**Pull request merged**